src/main/java/com/google/devtools/build/lib/remote/RemoteCache.java

```diff
@@ -130,16 +131,16 @@ public ActionResult downloadActionResult(
    */
   public ActionResult upload(
```

Comment:
My understanding is that the lifetime of a RemoteCache the a single command. Then why does one need to pass in a `RemotePathResolver` as opposed to making it a member of the class? (same for a number of other places below)

Reply:
`RemotePathResolver` is a drop in replacement for the usages of `execRoot` inside remote module. The question then becomes why did we pass in `execRoot` instead of making it a member of `RemoteCache`. Probably due to the time when we construct `RemoteCache` (inside `beforeCommand`), `execRoot` is not available yet. I can change this in another PR but let's make this PR only focus on changes for output paths semantics.

Reply:
...or else `execRoot` just ended up being there without any rhyme or reason. We'd have to take a look at the change history to tell, but it doesn't really matter, so meh. 

I think the question is more important for `RemotePathResolver` because it's an object with potentially a lot of state, unlike `Path`, which is just a value object. That said, if you want to clean it up later, all the power to you.